lib/streamlit/widgets.py

```diff
@@ -178,23 +199,21 @@ def get_widget_value(self, widget_id: str) -> Optional[Any]:
         if wstate is None:
             return None
 
-        value_type = wstate.WhichOneof("value")
-        if value_type == "json_value":
-            return json.loads(getattr(wstate, value_type))
-
-        return getattr(wstate, value_type)
+        return wstate.value()
 
     def set_state(self, widget_states: WidgetStates) -> None:
         """Copy the state from a WidgetStates protobuf into our state dict."""
         self._state = {}
         for wstate in widget_states.widgets:
-            self._state[wstate.id] = wstate
+            widget = Widget(wstate.id, wstate)
```

Comment:
Is this constructor one that we get for free from `attrs`? No strong opinion on it, but we could probably get rid of the manually written `__init__` method if this one will be used over it

Reply:
It isn't, because I wrote it as equivalent to what would be generated, but then removed the id argument from the constructor to make it less error prone. We could replace the id field with a getter method and then use the generated init.